herddb-core/src/main/java/herddb/core/TableManager.java

```diff
@@ -797,10 +797,10 @@ private long flushNewPage(DataPage page, Map<Bytes, Record> spareData) {
         boolean add = true;
         final Iterator<Record> records = spareData.values().iterator();
         while (add && records.hasNext()) {
-            Record record = records.next();
+            Record record = records.next().nonShared();
             add = page.put(record);
```

Comment:
This is going to allocate a new Record object

Reply:
Record is a wrapper around to Bytes. This will allocate a new record if bytes are shared.
Otherwise we sould set non final key and value inside record and replace them.